libs/librepcb/library/pkg/footprintpad.cpp

```diff
@@ -218,12 +218,26 @@ bool FootprintPad::setHeight(const PositiveLength& height) noexcept {
   return true;
 }
 
-bool FootprintPad::setDrillDiameter(const UnsignedLength& diameter) noexcept {
-  if (diameter == mDrillDiameter) {
-    return false;
+bool FootprintPad::setDrillSize(
+    const tl::optional<DrillSize>& drillSize) noexcept {
+  if (drillSize) {
+    if (mDrillSize) {
+      if (mDrillSize->getWidth() == drillSize->getWidth() &&
+          mDrillSize->getHeight() == drillSize->getHeight()) {
+        return false;
+      } else {
+        mDrillSize->setWidth(drillSize->getWidth());
+        mDrillSize->setHeight(drillSize->getHeight());
+      }
+    } else {
+      mDrillSize = DrillSize(*drillSize);
+    }
+  } else {
+    if (mDrillSize)
+      mDrillSize = tl::nullopt;
+    else
+      return false;
   }
```

Comment:
This behavior is what the code was doing before my most recent change, I simply inlined the setDrillSize I deleted.
I didn't know there was a sensible == for tl::optional, but since there is, this code is indeed unnecessarily complicated.

Reply:
Okay so if I'm reading https://github.com/TartanLlama/optional/blob/master/include/tl/optional.hpp#L1980 correctly, simply doing `mDrillSize = drillSize;` would make mDrillSize refer to the DrillSize in drillSize (if there is one).
Because drillSize is and mDrillSize is not const, this does not seem desirable - am I seeing this correctly?

Reply:
No, the link you posted is not the assignment operator used here. I think this is the correct one: https://github.com/TartanLlama/optional/blob/master/include/tl/optional.hpp#L1149

So it simply copies the contained value is there is one. There are no pointers or references used at all.

Reply:
Ah, sweet, that makes it a *lot* simpler